docs/source/index.rst

```diff
@@ -58,6 +58,7 @@ API Reference
    api/utils/avocado.utils.rst
    api/core/avocado.core.rst
    api/plugins/avocado.plugins.rst
+   api/optional-plugins/index
```

Comment:
This doesn't match what you described as done: "added the (optional) `.rst` suffix to match the style (at least at that paragraph)".  Actually, it looks like this commit is unchanged from v2.